### DISCUSSION TITLE
During inference with yolov8 segmentation model, does yolo resize the input to match the size of the training data? or is it fully convolutional and can take arbitrary size inputs.

### DISCUSSION THREAD

**Discussion by 7f0c** [ultralytics/ultralytics#9267] [Q&A]
During inference with yolov8 segmentation model, does yolo resize the input to match the size of the training data? or is it fully convolutional and can take arbitrary size inputs.

**Comment by glenn-jocher** [comment]
Hello!

Yes, during inference, the YOLOv8 segmentation model can take inputs of arbitrary sizes due to its fully convolutional nature. However, for optimal performance, it's common practice to resize inputs to match the size used during training, as this helps maintain the aspect ratio and ensures consistency. 🚀

```python
from ultralytics import YOLO

# Load model
model = YOLO('yolov8-seg.pt')

# Inference with an image (resizing to model's training size is advised but not mandatory)
results = model('path/to/your/image.jpg')
```

Feel free to adjust according to your specific needs!

**Reply by 7f0c** [reply]
is there a way to go the fully convolutional route instead of resizing? I don't want it to resize the image.

**Reply by glenn-jocher** [reply]
Hi there! 👋

Absolutely, YOLOv8 models are fully convolutional, so they can indeed handle arbitrary input sizes without resizing. You can pass your images directly to the model without specifying an image size. Here's how you can do it in code:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8-seg.pt')  # or any other pretrained model

# Predict without resizing
results = model('path/to/your/image.jpg')
```

Just keep in mind that using significantly different sizes from what the model was trained on might affect performance. Happy coding! 😄
